### PR TITLE
Feat: add python to auto detect for `--all-projects`

### PR DESCRIPTION
**Pull request by aviadhahami**
- [x] Ready for review
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
Adding `Pipfile` and `requirements.txt` to auto-detection w/ `--all-projects`

#### What are the relevant tickets?
[BST-1112](https://snyksec.atlassian.net/browse/BST-1112)

### CODE REVIEW

File: src/lib/find-files.ts [hunk 1]
```diff
@@ -188,11 +191,19 @@ function chooseBestManifest(
       return packageJson.path;
     }
     case 'rubygems': {
+      debug('Encountered multiple pip manifest files, defaulting to "Gemfile.lock"');
```
Comment (gitphill) [inline src/lib/find-files.ts]: typo? Encountered multiple Gemfiles...

File: Pipfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+name = "pypi"
+url = "https://pypi.org/simple"
+verify_ssl = true
+
+[dev-packages]
+
+[packages]
+
+[requires]
+python_version = "2.7"
```
Comment (gitphill) [inline Pipfile]: remove this file? or should it be in test fixtures?

File: test/acceptance/cli-test/cli-test.all-projects.spec.ts [hunk 3]
```diff
@@ -120,278 +141,278 @@ export const AllProjectsTests: AcceptanceTests = {
...
-          res,
-          `Target file:       test${path.sep}nuget-app-4${path.sep}packages.config`,
-          'Nuget project targetFile is as expected',
-        );
-        t.match(res, 'Package manager:   nuget', 'Nuget package manager');
-      } catch (err) {
-        t.fail('expected to pass');
-      }
-    },
+    //
```
Comment (gitphill) [inline test/acceptance/cli-test/cli-test.all-projects.spec.ts]: remove comments?

File: test/acceptance/cli-monitor/cli-monitor.all-projects.spec.ts [hunk 4]
```diff
@@ -201,6 +209,30 @@ export const AllProjectsTests: AcceptanceTests = {
...
 
+      // Pip project
+
+      // Removing properties that are different for plugin and custom-plugin
+      // (pluginName, pluginRuntime)
+      delete pipAll.body.meta.pluginName;
+      delete requestsPip.body.meta.pluginName;
+      delete requestsPip.body.meta.pluginRuntime;
+
+      // Python plugin (actually "custom-auto-detect" plugin) does't return
```
Comment (lili2311) [inline test/acceptance/cli-monitor/cli-monitor.all-projects.spec.ts]: ` does't` => ` doesn't`

File: test/acceptance/cli-monitor/cli-monitor.all-projects.spec.ts [hunk 5]
```diff
@@ -145,8 +145,8 @@ export const AllProjectsTests: AcceptanceTests = {
         allProjects: true,
```
Comment (orsagie) [inline test/acceptance/cli-monitor/cli-monitor.all-projects.spec.ts]: Add `skipUnresolved: true` here so we don't fail on python plugin

File: test/acceptance/cli-monitor/cli-monitor.all-projects.spec.ts [hunk 6]
```diff
@@ -200,6 +208,28 @@ export const AllProjectsTests: AcceptanceTests = {
         'Same body for --all-projects and --file=package-lock.json',
       );
 
+      // Pip project
+
+      // Removing properties that are different for plugin and custom-plugin
+      // (pluginName, pluginRuntime)
+      delete pipAll.body.meta.pluginName;
+      delete requestsPip.body.meta.pluginName;
+      delete requestsPip.body.meta.pluginRuntime;
```
Comment (lili2311) [inline test/acceptance/cli-monitor/cli-monitor.all-projects.spec.ts]: this is no longer needed can remove as both meta are now `exactly` the same

File: test/acceptance/cli-monitor/cli-monitor.all-projects.spec.ts [hunk 7]
```diff
@@ -200,6 +208,28 @@ export const AllProjectsTests: AcceptanceTests = {
...
+
+      // Python plugin (actually "custom-auto-detect" plugin) doesn't return
+      // "targetFile" prop as expected, so we picked it from another part of
+      // the req;
+      const {
+        context: {
+          package: { targetFile: pipTargetFile },
+        },
+      } = pipAll;
```
Comment (lili2311) [inline test/acceptance/cli-monitor/cli-monitor.all-projects.spec.ts]: same here, please don't mutate just compare the raw `pipAll.body` with `pipAll.body`

File: test/acceptance/cli-test/cli-test.all-projects.spec.ts [hunk 8]
```diff
@@ -102,6 +115,14 @@ export const AllProjectsTests: AcceptanceTests = {
       });
       const { body: mavenFileBody } = params.server.popRequest();
 
+      // Forcing targetFile into the payload as the new plugin doesn't return
```
Comment (lili2311) [inline test/acceptance/cli-test/cli-test.all-projects.spec.ts]: please remove the forcing of tragetFile

File: src/lib/detect.ts [hunk 9]
```diff
@@ -41,6 +41,8 @@ export const AUTO_DETECTABLE_FILES: string[] = [
   'packages.config',
   'project.json',
   'project.assets.json',
+  'Pipfile',
+  'requirements.txt',
```
Comment (lili2311) [inline src/lib/detect.ts]: is there a test for `requirements.txt`?

File: test/acceptance/cli-monitor/cli-monitor.all-projects.spec.ts [hunk 10]
```diff
@@ -9,7 +9,7 @@ interface AcceptanceTests {
 }
 
 export const AllProjectsTests: AcceptanceTests = {
-  language: 'Mixed (Ruby & Npm & Maven)',
+  language: 'Mixed (Ruby & Npm & Maven & Pip)',
```
Comment (gitphill) [inline test/acceptance/cli-monitor/cli-monitor.all-projects.spec.ts]: have just made this 'Mixed' on my go PR

File: src/lib/find-files.ts [hunk 11]
```diff
@@ -188,6 +191,9 @@ function chooseBestManifest(
       return packageJson.path;
     }
     case 'rubygems': {
+      debug(
```
Comment (dkontorovskyy) [inline src/lib/find-files.ts]: Do we need it? If yes - let's do it for every ecosystem, not for `rubygems` and `pip` only

**Comment by dkontorovskyy** [comment]
Not really sure if `paket` fixtures refactor should be in here

**Pull request merged**